Fix build errors

**Describe the bug**
When building the dotnet application a lot of errors are shown.

**To Reproduce**
Steps to reproduce the behavior:
1. Do  a `dotnet build`

**Expected behavior**
There should be no errors

**Actual behavior**
There are a lot of errors

**Screenshots**
N/A

**Environment (please complete the following information):**
docker with dotnet

**Additional context**
N/A

`Microsoft (R) Build Engine version 16.0.450+ga8dc7f1d34 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 680.65 ms for /app/GirafRest.csproj.
Setup/Startup.cs(61,43): warning CS0168: The variable 'ex' is declared but never used [/app/GirafRest.csproj]
Setup/GirafExtensions.cs(98,13): warning CS0618: 'ConsoleLoggerExtensions.AddConsole(ILoggerFactory)' is obsolete: 'This method is obsolete and will be removed in a future version. The recommended alternative is AddConsole(this ILoggingBuilder builder).' [/app/GirafRest.csproj]
Setup/GirafExtensions.cs(98,13): warning CS0618: 'DebugLoggerFactoryExtensions.AddDebug(ILoggerFactory)' is obsolete: 'This method is obsolete and will be removed in a future version. The recommended alternative is AddDebug(this ILoggingBuilder builder).' [/app/GirafRest.csproj]
Controllers/UserController.cs(153,47): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/GirafRest.csproj]
Controllers/UserController.cs(173,42): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/GirafRest.csproj]
Controllers/StatusController.cs(40,37): warning CS0168: The variable 'ex' is declared but never used [/app/GirafRest.csproj]
Controllers/PictogramController.cs(248,55): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/GirafRest.csproj]
Controllers/PictogramController.cs(287,45): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/GirafRest.csproj]
  GirafRest -> /app/bin/Release/netcoreapp2.2/GirafRest.dll
  GirafRest -> /app/out/
`

--- a/GirafRest.Test/GirafRest.Test.csproj
+++ b/GirafRest.Test/GirafRest.Test.csproj
@@ -10,14 +10,13 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="System.ComponentModel.TypeConverter" Version="4.3.*" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
     <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="2.2.0" />
     <PackageReference Include="xunit" Version="2.2.0" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
     <PackageReference Include="Moq" Version="4.7.8" />
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.2.0" />
-  </ItemGroup>
-  <ItemGroup>
     <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.6.1" />
     <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.Cookies" Version="2.2.0">
@@ -37,10 +36,6 @@
     <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.2.3" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="2.2.3">
     </PackageReference>
-  </ItemGroup>
-  <ItemGroup>
-    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="1.0.0-msbuild3-final" />
-    <DotNetCliToolReference Include="Microsoft.Extensions.SecretManager.Tools" Version="1.0.0-msbuild3-final" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="../GirafRest/GirafRest.csproj" />

--- a/GirafRest/Controllers/StatusController.cs
+++ b/GirafRest/Controllers/StatusController.cs
@@ -37,7 +37,7 @@ namespace GirafRest.Controllers
                 _giraf._context.Users.FirstOrDefault();
                 return new Response();
             }
-            catch (System.Exception ex)
+            catch (System.Exception)
             {
                 return new ErrorResponse(ErrorCode.Error);
             }

--- a/GirafRest/Controllers/UserController.cs
+++ b/GirafRest/Controllers/UserController.cs
@@ -150,17 +150,17 @@ namespace GirafRest.Controllers
         /// <returns>The requested image as a <see cref="ImageDTO"/></returns>
         /// <param name="id">Identifier of the <see cref="GirafUser"/>to get UserIcon for</param>
         [HttpGet("{id}/icon")]
-        public async Task<Response<ImageDTO>> GetUserIcon(string id)
+        public Task<Response<ImageDTO>> GetUserIcon(string id)
         {
             var user = _giraf._context.Users.Include(u => u.UserIcon).FirstOrDefault(u => u.Id == id);
             if (user == null)
-                return new ErrorResponse<ImageDTO>(ErrorCode.UserNotFound);
+                return Task.FromResult<Response<ImageDTO>>(new ErrorResponse<ImageDTO>(ErrorCode.UserNotFound));
                 
 
             if (user.UserIcon == null)
-                return new ErrorResponse<ImageDTO>(ErrorCode.UserHasNoIcon);
+                return Task.FromResult<Response<ImageDTO>>(new ErrorResponse<ImageDTO>(ErrorCode.UserHasNoIcon));
 
-            return new Response<ImageDTO>(new ImageDTO(user.UserIcon));
+            return Task.FromResult(new Response<ImageDTO>(new ImageDTO(user.UserIcon)));
         }
 
         /// <summary>
@@ -169,17 +169,18 @@ namespace GirafRest.Controllers
         /// <returns>The user icon as a png</returns>
         /// <param name="id">Identifier of the <see cref="GirafUser"/> to get icon for</param>
         [HttpGet("{id}/icon/raw")]
-        public async Task<IActionResult> GetRawUserIcon(string id)
+        public Task<IActionResult> GetRawUserIcon(string id)
         {
             var user = _giraf._context.Users.Include(u => u.UserIcon).FirstOrDefault(u => u.Id == id);
 
             if (user == null)
-                return NotFound();
+                return Task.FromResult<IActionResult>(NotFound());
 
             if (user.UserIcon == null)
-                return NotFound();
+                return Task.FromResult<IActionResult>(NotFound());
 
-            return File(Convert.FromBase64String(System.Text.Encoding.UTF8.GetString(user.UserIcon)), "image/png");
+
+            return Task.FromResult<IActionResult>(File(Convert.FromBase64String(System.Text.Encoding.UTF8.GetString(user.UserIcon)), "image/png"));
         }
 
         /// <summary>

--- a/GirafRest/Controllers/UserController.cs
+++ b/GirafRest/Controllers/UserController.cs
@@ -153,7 +153,6 @@ namespace GirafRest.Controllers
         public async Task<Response<ImageDTO>> GetUserIcon(string id)
         {
             var user = _giraf._context.Users.Include(u => u.UserIcon).FirstOrDefault(u => u.Id == id);
-
             if (user == null)
                 return new ErrorResponse<ImageDTO>(ErrorCode.UserNotFound);
                 

--- a/GirafRest/GirafRest.csproj
+++ b/GirafRest/GirafRest.csproj
@@ -5,6 +5,7 @@
   <PropertyGroup>
     <UserSecretsId>aspnet-GirafRest-f1c1b260-a093-40a1-977d-e78e374504a6</UserSecretsId>
   </PropertyGroup>
+  <PropertyGroup> <GenerateDocumentationFile>true</GenerateDocumentationFile> </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
     <DocumentationFile>bin\Debug\netcoreapp2.0\GirafRest.xml</DocumentationFile>
     <NoWarn>1701;1702;1705;1591;1572;1571;1573;1587;1570;1574;1587</NoWarn>

--- a/GirafRest/Setup/GirafExtensions.cs
+++ b/GirafRest/Setup/GirafExtensions.cs
@@ -93,15 +93,10 @@ namespace GirafRest.Extensions
         /// </summary>
         /// <param name="app">A reference to the application builder, that is used to define the behaviour of the server.</param>
         /// <param name="loggerFactory">A reference to the loggerFactory that is used to define the behaviour of the loggers.</param>
-        public static void ConfigureLogging(this IApplicationBuilder app/*, ILoggerFactory loggerFactory*/)
+        public static void ConfigureLogging(this IApplicationBuilder app)
         {
-            /*loggerFactory
-                .AddConsole()
-                .AddDebug();*/
             if (ProgramOptions.LogToFile)
             {
-                //Save log files corresponding to the strings defined in Program.cs, in this case logs/log.txt
-                /*loggerFactory.AddFile(Path.Combine(ProgramOptions.LogDirectory, ProgramOptions.LogFilepath), LogLevel.Warning);*/
                 app.UseStaticFiles();
 
                 //Ensure that the folder for the log-files exists - create it if not.

--- a/GirafRest/Setup/GirafExtensions.cs
+++ b/GirafRest/Setup/GirafExtensions.cs
@@ -93,15 +93,15 @@ namespace GirafRest.Extensions
         /// </summary>
         /// <param name="app">A reference to the application builder, that is used to define the behaviour of the server.</param>
         /// <param name="loggerFactory">A reference to the loggerFactory that is used to define the behaviour of the loggers.</param>
-        public static void ConfigureLogging(this IApplicationBuilder app, ILoggerFactory loggerFactory)
+        public static void ConfigureLogging(this IApplicationBuilder app/*, ILoggerFactory loggerFactory*/)
         {
-            loggerFactory
+            /*loggerFactory
                 .AddConsole()
-                .AddDebug();
+                .AddDebug();*/
             if (ProgramOptions.LogToFile)
             {
                 //Save log files corresponding to the strings defined in Program.cs, in this case logs/log.txt
-                loggerFactory.AddFile(Path.Combine(ProgramOptions.LogDirectory, ProgramOptions.LogFilepath), LogLevel.Warning);
+                /*loggerFactory.AddFile(Path.Combine(ProgramOptions.LogDirectory, ProgramOptions.LogFilepath), LogLevel.Warning);*/
                 app.UseStaticFiles();
 
                 //Ensure that the folder for the log-files exists - create it if not.

--- a/GirafRest/Setup/Startup.cs
+++ b/GirafRest/Setup/Startup.cs
@@ -193,7 +193,6 @@ namespace GirafRest.Setup
             app.UseIpRateLimiting();
 
             //Configure logging for the application
-            //app.ConfigureLogging(loggerFactory);
             appLifetime.ApplicationStopped.Register(Log.CloseAndFlush);
 
             app.UseStatusCodePagesWithReExecute("/v1/Error", "?status={0}");

--- a/GirafRest/Setup/Startup.cs
+++ b/GirafRest/Setup/Startup.cs
@@ -58,7 +58,7 @@ namespace GirafRest.Setup
             builder.AddEnvironmentVariables();
             try {
                 Configuration = builder.Build();
-            } catch(FileNotFoundException ex) {
+            } catch(FileNotFoundException) {
                 Console.WriteLine("ERROR: Missing appsettings file");
                 Console.WriteLine("Exiting...");
                 System.Environment.Exit(1);
@@ -193,7 +193,7 @@ namespace GirafRest.Setup
             app.UseIpRateLimiting();
 
             //Configure logging for the application
-            app.ConfigureLogging(loggerFactory);
+            //app.ConfigureLogging(loggerFactory);
             appLifetime.ApplicationStopped.Register(Log.CloseAndFlush);
 
             app.UseStatusCodePagesWithReExecute("/v1/Error", "?status={0}");

